### PR TITLE
Support Jinja2 templating of secret parameters

### PR DESCRIPTION
**Pull request by glennmatthews**
### Toward: #541

This should be the last incremental PR atop #868.

- Add support for Jinja2 templating of secrets provider parameters, allowing you to do things like specify a whole family of device passwords stored as text files on disk as `"/opt/nautobot/device_passwords/{{ obj.site.slug }}/{{ obj.name }}.txt"` or a whole family of Git repository secrets stored in environment variables as `"GIT_TOKEN_{{ obj.slug | replace('-', '_') | upper }}"`.
  - This does change the secrets API a bit, so existing (beta) secret provider implementations have to be updated to account for this change.
- Restructure and refine the Secrets documentation, particularly accounting for the fact that per-model documentation files aren't supposed to cross-link to other documentation files due to limitations of the online-help renderer.
- Improve test coverage somewhat.

### CODE REVIEW

**Comment by glennmatthews** [comment]
TODO: add "Secrets Providers" to the installed-plugins detail view page.

File: nautobot/docs/core-functionality/secrets.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+- Any [Job](../additional-features/jobs.md) can potentially access your Secrets, and can trivially log a Secret value to its JobResult, where it may be visible to users, so only install Jobs that you trust, carefully limit which users are able to execute jobs and view job results, and be aware of the potential for privilege escalation resulting from careless or malicious logging.
+- Any [Git repository](../models/extras/gitrepository.md) can add new Jobs to your system, so be careful about which users you grant the ability to create/edit `GitRepository` records.
+- Any user with access to [`nautobot-server nbshell`](../administration/nautobot-shell.md) can execute arbitrary code, including accessing Secrets, and will be able to bypass any Nautobot permissions restrictions as well.
+
+### Using Object Permissions with Secrets
+
+!!! tip
+    In practice you will likely want to carefully restrict which users are allowed to define and edit Secrets, and may want to use object permissions to further restrict which specific Secrets they are allowed to make use of.
```
Comment (jathanism) [inline nautobot/docs/core-functionality/secrets.md:34]:
```suggestion
    In practice you will likely want to carefully restrict which users are allowed to define and edit Secrets, and may want to use object permissions to further restrict which specific Secrets they are allowed to utilize.
```

**Review by jathanism — APPROVED** [review]
Another quality hit from Glenn. I have some minor comments/suggestions but nothing show stopping.

File: mkdocs.yml [hunk 2]
```diff
@@ -66,6 +66,7 @@ nav:
         - Circuits: 'core-functionality/circuits.md'
         - Power Tracking: 'core-functionality/power.md'
         - Tenancy: 'core-functionality/tenancy.md'
+        - Secrets: 'core-functionality/secrets.md'
```
Comment (jathanism) [inline mkdocs.yml:69]: We're getting to the point where it makes me wonder if we should alphabetize this list of core functionality items?

Comment (glennmatthews) [reply mkdocs.yml:69]: Sure can do. I don't see any particular logic to the current ordering.

**Review by jathanism — APPROVED** [review]

**Pull request merged**